bundle/bundle-writer.go

```diff
@@ -0,0 +1,352 @@
+	if err != nil {
+		fmt.Printf("Error while downloading the templates.json. err: %s", err)
+		return ManifestBundle{}, err
+	}
+	// open the template.json file and start downloading the stencils
+	templateJson, err := os.Open(tjPathfile)
+	if err != nil {
+		return ManifestBundle{}, err
+	}
+
+	templatesJsonData, err := ioutil.ReadAll(templateJson)
+	if err != nil {
+		return ManifestBundle{}, err
+	}
+
+	var templJson map[string]interface{}
+	err = json.Unmarshal([]byte(templatesJsonData), &templJson)
+	if err != nil {
+		return ManifestBundle{}, err
+	}
+
+	var manifestStencils = make([]*BundleStencil, 0)
+
+	for i, data := range templJson {
+		if i =="templates" {
+			for _, stencils := range data.([]interface{}) {
+				stencil := stencils.(map[string]interface{})
+				if stencil["min_usage"].(float64) > 0 {
+					if stencil["context_type"] == "service" {
+						for _, service := range services {
+							manifestFile, manifestStencils, err = downloadAndAddStencil(
+								service.Name,
+								stencil,
+								manifestFile,
+								skycapFolder,
+								templateRepository,
+								branch,
+								manifestStencils)
+							// create entry in manifest file with formatted name
+							// download and rename stencil file
+						}
+					}else {
+						manifestFile, manifestStencils, err = downloadAndAddStencil(
+							"",
+							stencil,
+							manifestFile,
+							skycapFolder,
+							templateRepository,
+							branch,
+							manifestStencils)
+					}
+				}
+			}
+		}
+	}
+	var newTemplate BundleBaseTemplates
+	newTemplate.Repo = githubURL
+	newTemplate.Branch = branch
+	newTemplate.Stencils = manifestStencils
+
+	manifestFile.BaseTemplates =  append(manifestFile.BaseTemplates, &newTemplate)
+
+	return manifestFile, nil
+}
+
+
+func loadManifest() (ManifestBundle, error) {
+	var manifest ManifestBundle
+	manifest.Version = "1"
+	manifest.Metadata = nil
+	manifest.Uid = ""
+	manifest.Name = ""
+	manifest.StencilGroups = make([]*BundleStencilGroup, 0)
+	manifest.BaseTemplates = make([]*BundleBaseTemplates, 0)
+	manifest.Policies = make([]*BundlePolicy, 0)
+	manifest.Tags = make([]string, 0)
+	manifest.HelmReleases = make([]*BundleHelmRelease, 0)
+	manifest.Configurations = make([]string, 0)
+
+	return manifest, nil
+}
+
+func saveManifest(skycapFolder string, content ManifestBundle) error {
+	out, err := json.MarshalIndent(content, "", "  ")
+	if err != nil {
+		return err
+	}
+	manifestPath := filepath.Join(skycapFolder, "manifest.json")
+	return ioutil.WriteFile(manifestPath, out, 0600)
+}
+
+func saveEnvVars(prefix string, envVars map[string]string , manifestFile ManifestBundle, skycapFolder string) (ManifestBundle, error) {
+	filename := prefix+"-config"
+	vars_path := filepath.Join(filepath.Join(skycapFolder,"configurations"), prefix+"-config")
+	var fileOut string
+	for key, value := range envVars {
+		fileOut = fileOut+key+"="+value+"\n"
```

Comment:
it's much more readable and easier to debug if you add each item as a string in a `[]string` and then join them at the end with a `strings.Join(foo[:],"\n")`